x/gov/client/rest/query.go

```diff
@@ -356,6 +354,13 @@ func queryVotesOnProposalHandlerFn(clientCtx client.Context) http.HandlerFunc {
 
 		// For inactive proposals we must query the txs directly to get the votes
 		// as they're no longer in state.
+		params := types.NewQueryProposalVotesParams(proposalID, page, limit)
+
+		bz, err = clientCtx.LegacyAmino.MarshalJSON(params)
+		if rest.CheckBadRequestError(w, err) {
+			return
+		}
+
```

Comment:
I would move this block in the `else` statement as that's the only place we need it.

Reply:
Updated